apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/StorageClientImplTest.cs

```diff
@@ -69,5 +76,402 @@ public void GetEffectiveEncryptionKey_Invalid(EncryptionKey clientkey, Encryptio
+            where T : new() => RetryThenFail(requestProvider, clientAction, new T(), errorCode, retryCount);
+
+        // Bucket test cases BEGIN
+
+        [Fact]
+        public void GetBucket_NoRetry() =>
+            NoRetryHelper(service => service.Buckets.Get("bucket"), client => client.GetBucket("bucket"));
+
+        [Fact]
+        public void GetBucket_RetryOnce() =>
+            RetryOnceHelper(service => service.Buckets.Get("bucket"), client => client.GetBucket("bucket"));
+
+        [Fact]
+        public void GetBucket_RetryThenFail() =>
+            RetryThenFailHelper(service => service.Buckets.Get("bucket"), client => client.GetBucket("bucket"));
+
+        [Fact]
+        public void DeleteBucket_NoRetry() =>
+            NoRetryHelper(service => service.Buckets.Delete("bucket"), client => client.DeleteBucket("bucket"));
+
+        [Fact]
+        public void DeleteBucket_RetryOnce() =>
+            RetryOnce(service => service.Buckets.Delete("bucket"), client => client.DeleteBucket("bucket"), "check");
+
+        [Fact]
+        public void DeleteBucket_RetryThenFail() =>
+            RetryThenFail(service => service.Buckets.Delete("bucket"), client => client.DeleteBucket("bucket"), "check");
+
+        [Fact]
+        public void ListBucket_NoRetry() =>
+            NoRetryHelper(service => new BucketsResource.ListRequest(service, "project"), client => client.ListBuckets("project").Count());
+
+        [Fact]
+        public void ListBucket_RetryOnce() =>
+            RetryOnceHelper(service => service.Buckets.List("project"), client => client.ListBuckets("project").Count());
+
+        [Fact]
+        public void ListBucket_RetryThenFail() =>
+            RetryThenFailHelper(service => service.Buckets.List("project"), client => client.ListBuckets("project").Count());
+
+        [Fact]
+        public void InsertBucket_RetryOnce() =>
+            RetryOnceHelper(service => service.Buckets.Insert(new Bucket { Name = "bucket" }, "project"), client => client.CreateBucket("project", "bucket"));
+
+        [Fact]
+        public void InsertBucket_RetryThenFail() =>
+            RetryThenFailHelper(service => service.Buckets.Insert(new Bucket { Name = "bucket" }, "project"), client => client.CreateBucket("project", "bucket"));
+
+        [Fact]
+        public void InsertBucket_NoRetry() =>
+            NoRetryHelper(service => service.Buckets.Insert(new Bucket { Name = "bucket" }, "project"), client => client.CreateBucket("project", "bucket"));
+
+        [Fact]
+        public void UpdateBucket_WithMetageneration_RetryOnce() =>
+            RetryOnceHelper(service => new BucketsResource.UpdateRequest(service, new Bucket { Name = "bucket" }, "bucket") { IfMetagenerationMatch = 70 }, 
+                client => client.UpdateBucket(new Bucket { Name = "bucket" }, new UpdateBucketOptions() { IfMetagenerationMatch = 70 }));
+
+        [Fact]
+        public void UpdateBucket_withoutMetgeneration_NoRetry() =>
```

Comment:
Nit: without => Without and Metgeneration => Metageneration (ditto for PatchBucket test below)